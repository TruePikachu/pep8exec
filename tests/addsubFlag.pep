	; Do all 8-bit additions and subtractions
	LDA	0,i
	STA	n1,d
	STA	n2,d
loop:	LDA	n1,d
	ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	STA	a1,d
	LDA	n2,d
	ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	STA	a2,d
	LDX	a1,d
	ADDX	a2,d
	MOVFLGA
	STA	flg,d
	ASRX
	ASRX
	ASRX
	ASRX
	ASRX
	ASRX
	ASRX
	ASRX
	LDA	n1,d
	CALL	DByteA
	CHARO	'+',i
	LDA	n2,d
	CALL	DByteA
	CHARO	'=',i
	STX	tmp,d
	LDA	tmp,d
	CALL	DByteA
	CHARO	' ',i
	LDA	flg,d
	CALL	DispFlgA
	CHARO	'\n',i
	LDX	a1,d
	SUBX	a2,d
	MOVFLGA
	STA	flg,d
	ASRX
	ASRX
	ASRX
	ASRX
	ASRX
	ASRX
	ASRX
	ASRX
	LDA	n1,d
	CALL	DByteA
	CHARO	'-',i
	LDA	n2,d
	CALL	DByteA
	CHARO	'=',i
	STX	tmp,d
	LDA	tmp,d
	CALL	DByteA
	CHARO	' ',i
	LDA	flg,d
	CALL	DispFlgA
	CHARO	'\n',i
	LDA	n1,d
	ADDA	0x10,i
	ANDA	0xFF,i
	STA	n1,d
	BRNE	loop
	LDA	n2,d
	ADDA	0x10,i
	ANDA	0xFF,i
	STA	n2,d
	BRNE	loop
	STOP
flg:	.WORD	0
tmp:	.WORD	0
n1:	.WORD	0
n2:	.WORD	0
a1:	.WORD	0
a2:	.WORD	0

;;; PROCEDURE DispFlgA
DispFlgA:SUBSP	4,i
	STA	0,s
	STX	2,s
	LDX	0,i
	RORA
	RORA
	RORA
	RORA
DFAloop:ROLA
	BRC	DFAbit
	CHARO	' ',i
	BR	DFAmrge
DFAbit:	CHARO	DFAflgs,x
DFAmrge:ADDX	1,i
	CPX	4,i
	BRNE	DFAloop
	LDX	2,s
	LDA	0,s
	RET4
DFAflgs:.ASCII	"NZVC"
;;; END PROCEDURE

;;; PROCEDURE DByteA
DByteA:	SUBSP	4,i
	STA	0,s
	STX	2,s
	LDX	0,s
	ASRX
	ASRX
	ASRX
	ASRX
	ANDX	0x0F,i
	CHARO	DBAhex,x
	LDX	0,s
	ANDX	0x0F,i
	CHARO	DBAhex,x
	LDX	2,s
	RET4
DBAhex:	.ASCII	"0123456789ABCDEF"
;;; END PROCEDURE

;;; PROCEDURE SgnExtA
SgnExtA:ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	ASLA
	ASRA
	ASRA
	ASRA
	ASRA
	ASRA
	ASRA
	ASRA
	ASRA
	RET0
;;; END PROCEDURE
	.END
