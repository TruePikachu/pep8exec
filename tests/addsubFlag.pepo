C0 00 00 E1 00 BA E1 00 BC C1 00 BA 1C 1C 1C 1C
1C 1C 1C 1C E1 00 BE C1 00 BC 1C 1C 1C 1C 1C 1C
1C 1C E1 00 C0 C9 00 BE 79 00 C0 03 E1 00 B6 1F
1F 1F 1F 1F 1F 1F 1F C1 00 BA 16 00 F3 50 00 2B
C1 00 BC 16 00 F3 50 00 3D E9 00 B8 C1 00 B8 16
00 F3 50 00 20 C1 00 B6 16 00 C2 50 00 0A C9 00
BE 89 00 C0 03 E1 00 B6 1F 1F 1F 1F 1F 1F 1F 1F
C1 00 BA 16 00 F3 50 00 2D C1 00 BC 16 00 F3 50
00 3D E9 00 B8 C1 00 B8 16 00 F3 50 00 20 C1 00
B6 16 00 C2 50 00 0A C1 00 BA 70 00 10 90 00 FF
E1 00 BA 0C 00 09 C1 00 BC 70 00 10 90 00 FF E1
00 BC 0C 00 09 00 00 00 00 00 00 00 00 00 00 00
00 00 68 00 04 E3 00 00 EB 00 02 C8 00 00 22 22
22 22 20 14 00 DC 50 00 20 04 00 DF 55 00 EF 78
00 01 B8 00 04 0C 00 D2 CB 00 02 C3 00 00 5C 4E
5A 56 43 68 00 04 E3 00 00 EB 00 02 CB 00 00 1F
1F 1F 1F 98 00 0F 55 01 16 CB 00 00 98 00 0F 55
01 16 CB 00 02 5C 30 31 32 33 34 35 36 37 38 39
41 42 43 44 45 46 1C 1C 1C 1C 1C 1C 1C 1C 1E 1E
1E 1E 1E 1E 1E 1E 58 zz
