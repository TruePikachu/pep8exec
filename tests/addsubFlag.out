00+00=00  Z  
00-00=00  Z  
10+00=10     
10-00=10     
20+00=20     
20-00=20     
30+00=30     
30-00=30     
40+00=40     
40-00=40     
50+00=50     
50-00=50     
60+00=60     
60-00=60     
70+00=70     
70-00=70     
80+00=80 N   
80-00=80 N   
90+00=90 N   
90-00=90 N   
A0+00=A0 N   
A0-00=A0 N   
B0+00=B0 N   
B0-00=B0 N   
C0+00=C0 N   
C0-00=C0 N   
D0+00=D0 N   
D0-00=D0 N   
E0+00=E0 N   
E0-00=E0 N   
F0+00=F0 N   
F0-00=F0 N   
00+10=10     
00-10=F0 N  C
10+10=20     
10-10=00  Z  
20+10=30     
20-10=10     
30+10=40     
30-10=20     
40+10=50     
40-10=30     
50+10=60     
50-10=40     
60+10=70     
60-10=50     
70+10=80 N V 
70-10=60     
80+10=90 N   
80-10=70   V 
90+10=A0 N   
90-10=80 N   
A0+10=B0 N   
A0-10=90 N   
B0+10=C0 N   
B0-10=A0 N   
C0+10=D0 N   
C0-10=B0 N   
D0+10=E0 N   
D0-10=C0 N   
E0+10=F0 N   
E0-10=D0 N   
F0+10=00  Z C
F0-10=E0 N   
00+20=20     
00-20=E0 N  C
10+20=30     
10-20=F0 N  C
20+20=40     
20-20=00  Z  
30+20=50     
30-20=10     
40+20=60     
40-20=20     
50+20=70     
50-20=30     
60+20=80 N V 
60-20=40     
70+20=90 N V 
70-20=50     
80+20=A0 N   
80-20=60   V 
90+20=B0 N   
90-20=70   V 
A0+20=C0 N   
A0-20=80 N   
B0+20=D0 N   
B0-20=90 N   
C0+20=E0 N   
C0-20=A0 N   
D0+20=F0 N   
D0-20=B0 N   
E0+20=00  Z C
E0-20=C0 N   
F0+20=10    C
F0-20=D0 N   
00+30=30     
00-30=D0 N  C
10+30=40     
10-30=E0 N  C
20+30=50     
20-30=F0 N  C
30+30=60     
30-30=00  Z  
40+30=70     
40-30=10     
50+30=80 N V 
50-30=20     
60+30=90 N V 
60-30=30     
70+30=A0 N V 
70-30=40     
80+30=B0 N   
80-30=50   V 
90+30=C0 N   
90-30=60   V 
A0+30=D0 N   
A0-30=70   V 
B0+30=E0 N   
B0-30=80 N   
C0+30=F0 N   
C0-30=90 N   
D0+30=00  Z C
D0-30=A0 N   
E0+30=10    C
E0-30=B0 N   
F0+30=20    C
F0-30=C0 N   
00+40=40     
00-40=C0 N  C
10+40=50     
10-40=D0 N  C
20+40=60     
20-40=E0 N  C
30+40=70     
30-40=F0 N  C
40+40=80 N V 
40-40=00  Z  
50+40=90 N V 
50-40=10     
60+40=A0 N V 
60-40=20     
70+40=B0 N V 
70-40=30     
80+40=C0 N   
80-40=40   V 
90+40=D0 N   
90-40=50   V 
A0+40=E0 N   
A0-40=60   V 
B0+40=F0 N   
B0-40=70   V 
C0+40=00  Z C
C0-40=80 N   
D0+40=10    C
D0-40=90 N   
E0+40=20    C
E0-40=A0 N   
F0+40=30    C
F0-40=B0 N   
00+50=50     
00-50=B0 N  C
10+50=60     
10-50=C0 N  C
20+50=70     
20-50=D0 N  C
30+50=80 N V 
30-50=E0 N  C
40+50=90 N V 
40-50=F0 N  C
50+50=A0 N V 
50-50=00  Z  
60+50=B0 N V 
60-50=10     
70+50=C0 N V 
70-50=20     
80+50=D0 N   
80-50=30   V 
90+50=E0 N   
90-50=40   V 
A0+50=F0 N   
A0-50=50   V 
B0+50=00  Z C
B0-50=60   V 
C0+50=10    C
C0-50=70   V 
D0+50=20    C
D0-50=80 N   
E0+50=30    C
E0-50=90 N   
F0+50=40    C
F0-50=A0 N   
00+60=60     
00-60=A0 N  C
10+60=70     
10-60=B0 N  C
20+60=80 N V 
20-60=C0 N  C
30+60=90 N V 
30-60=D0 N  C
40+60=A0 N V 
40-60=E0 N  C
50+60=B0 N V 
50-60=F0 N  C
60+60=C0 N V 
60-60=00  Z  
70+60=D0 N V 
70-60=10     
80+60=E0 N   
80-60=20   V 
90+60=F0 N   
90-60=30   V 
A0+60=00  Z C
A0-60=40   V 
B0+60=10    C
B0-60=50   V 
C0+60=20    C
C0-60=60   V 
D0+60=30    C
D0-60=70   V 
E0+60=40    C
E0-60=80 N   
F0+60=50    C
F0-60=90 N   
00+70=70     
00-70=90 N  C
10+70=80 N V 
10-70=A0 N  C
20+70=90 N V 
20-70=B0 N  C
30+70=A0 N V 
30-70=C0 N  C
40+70=B0 N V 
40-70=D0 N  C
50+70=C0 N V 
50-70=E0 N  C
60+70=D0 N V 
60-70=F0 N  C
70+70=E0 N V 
70-70=00  Z  
80+70=F0 N   
80-70=10   V 
90+70=00  Z C
90-70=20   V 
A0+70=10    C
A0-70=30   V 
B0+70=20    C
B0-70=40   V 
C0+70=30    C
C0-70=50   V 
D0+70=40    C
D0-70=60   V 
E0+70=50    C
E0-70=70   V 
F0+70=60    C
F0-70=80 N   
00+80=80 N   
00-80=80 N VC
10+80=90 N   
10-80=90 N VC
20+80=A0 N   
20-80=A0 N VC
30+80=B0 N   
30-80=B0 N VC
40+80=C0 N   
40-80=C0 N VC
50+80=D0 N   
50-80=D0 N VC
60+80=E0 N   
60-80=E0 N VC
70+80=F0 N   
70-80=F0 N VC
80+80=00  ZVC
80-80=00  Z  
90+80=10   VC
90-80=10     
A0+80=20   VC
A0-80=20     
B0+80=30   VC
B0-80=30     
C0+80=40   VC
C0-80=40     
D0+80=50   VC
D0-80=50     
E0+80=60   VC
E0-80=60     
F0+80=70   VC
F0-80=70     
00+90=90 N   
00-90=70    C
10+90=A0 N   
10-90=80 N VC
20+90=B0 N   
20-90=90 N VC
30+90=C0 N   
30-90=A0 N VC
40+90=D0 N   
40-90=B0 N VC
50+90=E0 N   
50-90=C0 N VC
60+90=F0 N   
60-90=D0 N VC
70+90=00  Z C
70-90=E0 N VC
80+90=10   VC
80-90=F0 N  C
90+90=20   VC
90-90=00  Z  
A0+90=30   VC
A0-90=10     
B0+90=40   VC
B0-90=20     
C0+90=50   VC
C0-90=30     
D0+90=60   VC
D0-90=40     
E0+90=70   VC
E0-90=50     
F0+90=80 N  C
F0-90=60     
00+A0=A0 N   
00-A0=60    C
10+A0=B0 N   
10-A0=70    C
20+A0=C0 N   
20-A0=80 N VC
30+A0=D0 N   
30-A0=90 N VC
40+A0=E0 N   
40-A0=A0 N VC
50+A0=F0 N   
50-A0=B0 N VC
60+A0=00  Z C
60-A0=C0 N VC
70+A0=10    C
70-A0=D0 N VC
80+A0=20   VC
80-A0=E0 N  C
90+A0=30   VC
90-A0=F0 N  C
A0+A0=40   VC
A0-A0=00  Z  
B0+A0=50   VC
B0-A0=10     
C0+A0=60   VC
C0-A0=20     
D0+A0=70   VC
D0-A0=30     
E0+A0=80 N  C
E0-A0=40     
F0+A0=90 N  C
F0-A0=50     
00+B0=B0 N   
00-B0=50    C
10+B0=C0 N   
10-B0=60    C
20+B0=D0 N   
20-B0=70    C
30+B0=E0 N   
30-B0=80 N VC
40+B0=F0 N   
40-B0=90 N VC
50+B0=00  Z C
50-B0=A0 N VC
60+B0=10    C
60-B0=B0 N VC
70+B0=20    C
70-B0=C0 N VC
80+B0=30   VC
80-B0=D0 N  C
90+B0=40   VC
90-B0=E0 N  C
A0+B0=50   VC
A0-B0=F0 N  C
B0+B0=60   VC
B0-B0=00  Z  
C0+B0=70   VC
C0-B0=10     
D0+B0=80 N  C
D0-B0=20     
E0+B0=90 N  C
E0-B0=30     
F0+B0=A0 N  C
F0-B0=40     
00+C0=C0 N   
00-C0=40    C
10+C0=D0 N   
10-C0=50    C
20+C0=E0 N   
20-C0=60    C
30+C0=F0 N   
30-C0=70    C
40+C0=00  Z C
40-C0=80 N VC
50+C0=10    C
50-C0=90 N VC
60+C0=20    C
60-C0=A0 N VC
70+C0=30    C
70-C0=B0 N VC
80+C0=40   VC
80-C0=C0 N  C
90+C0=50   VC
90-C0=D0 N  C
A0+C0=60   VC
A0-C0=E0 N  C
B0+C0=70   VC
B0-C0=F0 N  C
C0+C0=80 N  C
C0-C0=00  Z  
D0+C0=90 N  C
D0-C0=10     
E0+C0=A0 N  C
E0-C0=20     
F0+C0=B0 N  C
F0-C0=30     
00+D0=D0 N   
00-D0=30    C
10+D0=E0 N   
10-D0=40    C
20+D0=F0 N   
20-D0=50    C
30+D0=00  Z C
30-D0=60    C
40+D0=10    C
40-D0=70    C
50+D0=20    C
50-D0=80 N VC
60+D0=30    C
60-D0=90 N VC
70+D0=40    C
70-D0=A0 N VC
80+D0=50   VC
80-D0=B0 N  C
90+D0=60   VC
90-D0=C0 N  C
A0+D0=70   VC
A0-D0=D0 N  C
B0+D0=80 N  C
B0-D0=E0 N  C
C0+D0=90 N  C
C0-D0=F0 N  C
D0+D0=A0 N  C
D0-D0=00  Z  
E0+D0=B0 N  C
E0-D0=10     
F0+D0=C0 N  C
F0-D0=20     
00+E0=E0 N   
00-E0=20    C
10+E0=F0 N   
10-E0=30    C
20+E0=00  Z C
20-E0=40    C
30+E0=10    C
30-E0=50    C
40+E0=20    C
40-E0=60    C
50+E0=30    C
50-E0=70    C
60+E0=40    C
60-E0=80 N VC
70+E0=50    C
70-E0=90 N VC
80+E0=60   VC
80-E0=A0 N  C
90+E0=70   VC
90-E0=B0 N  C
A0+E0=80 N  C
A0-E0=C0 N  C
B0+E0=90 N  C
B0-E0=D0 N  C
C0+E0=A0 N  C
C0-E0=E0 N  C
D0+E0=B0 N  C
D0-E0=F0 N  C
E0+E0=C0 N  C
E0-E0=00  Z  
F0+E0=D0 N  C
F0-E0=10     
00+F0=F0 N   
00-F0=10    C
10+F0=00  Z C
10-F0=20    C
20+F0=10    C
20-F0=30    C
30+F0=20    C
30-F0=40    C
40+F0=30    C
40-F0=50    C
50+F0=40    C
50-F0=60    C
60+F0=50    C
60-F0=70    C
70+F0=60    C
70-F0=80 N VC
80+F0=70   VC
80-F0=90 N  C
90+F0=80 N  C
90-F0=A0 N  C
A0+F0=90 N  C
A0-F0=B0 N  C
B0+F0=A0 N  C
B0-F0=C0 N  C
C0+F0=B0 N  C
C0-F0=D0 N  C
D0+F0=C0 N  C
D0-F0=E0 N  C
E0+F0=D0 N  C
E0-F0=F0 N  C
F0+F0=E0 N  C
F0-F0=00  Z  
